[2024-11-22T07:36:01.210+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-22T07:36:01.216+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: weather_stock_prediction.predict_prices manual__2024-11-22T07:35:47.494208+00:00 [queued]>
[2024-11-22T07:36:01.219+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: weather_stock_prediction.predict_prices manual__2024-11-22T07:35:47.494208+00:00 [queued]>
[2024-11-22T07:36:01.219+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 2
[2024-11-22T07:36:01.223+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): predict_prices> on 2024-11-22 07:35:47.494208+00:00
[2024-11-22T07:36:01.227+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=11400) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-22T07:36:01.228+0000] {standard_task_runner.py:72} INFO - Started process 11411 to run task
[2024-11-22T07:36:01.228+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'weather_stock_prediction', 'predict_prices', 'manual__2024-11-22T07:35:47.494208+00:00', '--job-id', '959', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmpbmetgxmd']
[2024-11-22T07:36:01.229+0000] {standard_task_runner.py:105} INFO - Job 959: Subtask predict_prices
[2024-11-22T07:36:01.250+0000] {task_command.py:467} INFO - Running <TaskInstance: weather_stock_prediction.predict_prices manual__2024-11-22T07:35:47.494208+00:00 [running]> on host cba1e82f0ade
[2024-11-22T07:36:01.293+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='weather_stock_prediction' AIRFLOW_CTX_TASK_ID='predict_prices' AIRFLOW_CTX_EXECUTION_DATE='2024-11-22T07:35:47.494208+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-22T07:35:47.494208+00:00'
[2024-11-22T07:36:01.294+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-11-22T07:36:01.294+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-11-22T07:36:01.294+0000] {logging_mixin.py:190} INFO - Current task name:predict_prices state:running start_date:2024-11-22 07:36:01.216613+00:00
[2024-11-22T07:36:01.295+0000] {logging_mixin.py:190} INFO - Dag name:weather_stock_prediction and current dag run status:running
[2024-11-22T07:36:01.295+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-22T07:36:01.301+0000] {logging_mixin.py:190} INFO - Weather forecast data: {'date': {'0': '2024-11-21', '1': '2024-11-22', '2': '2024-11-23', '3': '2024-11-24', '4': '2024-11-25', '5': '2024-11-26', '6': '2024-11-27', '7': '2024-11-28', '8': '2024-11-29', '9': '2024-11-30', '10': '2024-12-01', '11': '2024-12-02', '12': '2024-12-03', '13': '2024-12-04', '14': '2024-12-05', '15': '2024-12-06'}, 'temperature': {'0': 15.35, '1': 14.94, '2': 13.3, '3': 12.75, '4': 12.84, '5': 12.23, '6': 11.72, '7': 10.73, '8': 10.74, '9': 11.19, '10': 11.87, '11': 11.96, '12': 12.07, '13': 15.04, '14': 15.66, '15': 15.9}, 'min_temperature': {'0': 14.22, '1': 12.33, '2': 11.05, '3': 10.15, '4': 11.19, '5': 11.52, '6': 10.1, '7': 9.12, '8': 8.87, '9': 9.07, '10': 9.72, '11': 10.39, '12': 10.02, '13': 10.47, '14': 11.17, '15': 11.47}, 'max_temperature': {'0': 15.95, '1': 15.86, '2': 13.41, '3': 13.37, '4': 14.17, '5': 13.06, '6': 13.46, '7': 13.75, '8': 14.05, '9': 14.47, '10': 14.95, '11': 14.52, '12': 15.0, '13': 15.55, '14': 15.66, '15': 16.08}, 'humidity': {'0': 76, '1': 85, '2': 74, '3': 61, '4': 69, '5': 83, '6': 57, '7': 58, '8': 57, '9': 57, '10': 55, '11': 54, '12': 54, '13': 47, '14': 45, '15': 49}, 'wind_speed': {'0': 10.89, '1': 9.04, '2': 4.91, '3': 6.3, '4': 6.67, '5': 4.29, '6': 3.82, '7': 3.96, '8': 3.99, '9': 4.05, '10': 4.15, '11': 3.48, '12': 3.66, '13': 3.74, '14': 3.19, '15': 2.99}}
[2024-11-22T07:36:01.317+0000] {logging_mixin.py:190} INFO - DataFrame structure:                                                 date  ...                                         wind_speed
0  {'0': '2024-11-21', '1': '2024-11-22', '2': '2...  ...  {'0': 10.89, '1': 9.04, '2': 4.91, '3': 6.3, '...

[1 rows x 6 columns]
[2024-11-22T07:36:01.320+0000] {logging_mixin.py:190} INFO - Cleaned DataFrame:    date  temperature  min_temperature  max_temperature  humidity  wind_speed
0   NaN          NaN              NaN              NaN       NaN         NaN
[2024-11-22T07:36:01.321+0000] {logging_mixin.py:190} INFO - DataFrame contains missing values. Applying imputation.
[2024-11-22T07:36:01.324+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/impute/_base.py:598: UserWarning: Skipping features without any observed values: ['date' 'temperature' 'min_temperature' 'max_temperature' 'humidity'
 'wind_speed']. At least one non-missing value is needed for imputation with strategy='mean'.
  warnings.warn(

[2024-11-22T07:36:01.324+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/dag.py", line 296, in predict_prices
    df = pd.DataFrame(imputer.fit_transform(df), columns=df.columns)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/frame.py", line 782, in __init__
    mgr = ndarray_to_mgr(
          ^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/internals/construction.py", line 336, in ndarray_to_mgr
    _check_values_indices_shape_match(values, index, columns)
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/internals/construction.py", line 420, in _check_values_indices_shape_match
    raise ValueError(f"Shape of passed values is {passed}, indices imply {implied}")
ValueError: Shape of passed values is (1, 0), indices imply (1, 6)
[2024-11-22T07:36:01.332+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-11-22T07:36:01.332+0000] {logging_mixin.py:190} INFO - Task start:2024-11-22 07:36:01.216613+00:00 end:2024-11-22 07:36:01.332190+00:00 duration:0.115577
[2024-11-22T07:36:01.332+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): predict_prices> dag:<DAG: weather_stock_prediction> dagrun:<DagRun weather_stock_prediction @ 2024-11-22 07:35:47.494208+00:00: manual__2024-11-22T07:35:47.494208+00:00, state:running, queued_at: 2024-11-22 07:35:47.521335+00:00. externally triggered: True>
[2024-11-22T07:36:01.333+0000] {logging_mixin.py:190} INFO - Failure caused by Shape of passed values is (1, 0), indices imply (1, 6)
[2024-11-22T07:36:01.333+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=weather_stock_prediction, task_id=predict_prices, run_id=manual__2024-11-22T07:35:47.494208+00:00, execution_date=20241122T073547, start_date=20241122T073601, end_date=20241122T073601
[2024-11-22T07:36:01.339+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-22T07:36:01.339+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 959 for task predict_prices (Shape of passed values is (1, 0), indices imply (1, 6); 11411)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/dag.py", line 296, in predict_prices
    df = pd.DataFrame(imputer.fit_transform(df), columns=df.columns)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/frame.py", line 782, in __init__
    mgr = ndarray_to_mgr(
          ^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/internals/construction.py", line 336, in ndarray_to_mgr
    _check_values_indices_shape_match(values, index, columns)
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/core/internals/construction.py", line 420, in _check_values_indices_shape_match
    raise ValueError(f"Shape of passed values is {passed}, indices imply {implied}")
ValueError: Shape of passed values is (1, 0), indices imply (1, 6)
[2024-11-22T07:36:01.375+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-11-22T07:36:01.384+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-11-22T07:36:01.385+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
